components/director/internal/domain/runtime/resolver.go

```diff
@@ -3,49 +3,190 @@ package runtime
 import (
 	"context"
 
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+
+	"github.com/kyma-incubator/compass/components/director/internal/labelfilter"
+
 	"github.com/kyma-incubator/compass/components/director/internal/graphql"
 )
 
-type svc interface{}
+//go:generate mockery -name=RuntimeService -output=automock -outpkg=automock -case=underscore
+type RuntimeService interface {
+	Create(ctx context.Context, in model.RuntimeInput) (string, error)
+	Update(ctx context.Context, id string, in model.RuntimeInput) error
+	Get(ctx context.Context, id string) (*model.Runtime, error)
+	Delete(ctx context.Context, id string) error
+	List(ctx context.Context, filter []*labelfilter.LabelFilter, pageSize *int, cursor *string) (*RuntimePage, error)
+	AddLabel(ctx context.Context, runtimeID string, key string, values []string) error
+	DeleteLabel(ctx context.Context, runtimeID string, key string, values []string) error
+	AddAnnotation(ctx context.Context, runtimeID string, key string, value interface{}) error
+	DeleteAnnotation(ctx context.Context, runtimeID string, key string) error
+}
+
+//go:generate mockery -name=RuntimeConverter -output=automock -outpkg=automock -case=underscore
+type RuntimeConverter interface {
+	ToGraphQL(in *model.Runtime) *graphql.Runtime
+	MultipleToGraphQL(in []*model.Runtime) []*graphql.Runtime
+	InputFromGraphQL(in graphql.RuntimeInput) model.RuntimeInput
+}
 
 type Resolver struct {
-	svc       svc
-	converter *Converter
+	svc       RuntimeService
+	converter RuntimeConverter
 }
 
-func NewResolver(svc svc) *Resolver {
+func NewResolver(svc RuntimeService) *Resolver {
 	return &Resolver{
 		svc:       svc,
 		converter: &Converter{},
 	}
 }
 
+// TODO: Proper error handling
+// TODO: Pagination
+
 func (r *Resolver) Runtimes(ctx context.Context, filter []*graphql.LabelFilter, first *int, after *graphql.PageCursor) (*graphql.RuntimePage, error) {
-	panic("not implemented")
+	labelFilter := labelfilter.MultipleFromGraphQL(filter)
+
+	var cursor string
+	if after != nil {
+		cursor = string(*after)
+	}
+
+	runtimesPage, err := r.svc.List(ctx, labelFilter, first, &cursor)
+	if err != nil {
+		return nil, err
```

Comment:
in CBS we have a special layer of errors, that are displayed for end-user. Why we don't follow that pattern?
Also, why we don't wrap errors?

Reply:
Here's an issue for that: https://github.com/kyma-incubator/compass/issues/66

It will be done as a follow-up task.